tasks/kernel_matrix_testing/compiler.py

```diff
@@ -76,6 +76,12 @@ def start(self) -> None:
         if self.is_running:
             self.stop()
 
+        info("[+] Updating build image, ensuring we can log in to the registry")
+        self.ctx.run(
+            "aws-vault exec sso-build-stable-developer -- aws ecr --region us-east-1 get-login-password | docker login --username AWS --password-stdin 486234852809.dkr.ecr.us-east-1.amazonaws.com"
+        )
+        self.ctx.run(f"docker pull {self.image}")
```

Comment:
Do you think it makes sense to try pulling first, and then logging in if that fails? Or maybe checking if the image exists locally before logging in and trying to pull it?

Reply:
Maybe checking if the image exists is the best option, as for a given tag the image shouldn't change. Logging in only after a failure is a worse option ebcause you can't really know why the pull failed reliably.